### PR TITLE
ECOM-7021 Add tags to course run serializer

### PR DESCRIPTION
**Pull request by MatthewPiatetsky**
@edx/ecommerce

### CODE REVIEW

File: course_discovery/apps/api/tests/test_serializers.py [hunk 1]
```diff
@@ -294,11 +295,13 @@ def test_data(self):
         course_run = CourseRunFactory()
         serializer = self.serializer_class(course_run, context={'request': request})
         expected = self.get_expected_data(course_run, request)
+        course_run.tags.add(self.TAG)
```
Comment (clintonb) [inline course_discovery/apps/api/tests/test_serializers.py]: This is not necessary.

File: course_discovery/apps/api/tests/test_serializers.py [hunk 2]
```diff
@@ -266,6 +266,7 @@ def test_published_course_runs_only(self, published_course_runs_only):
 
 class MinimalCourseRunSerializerTests(TestCase):
     serializer_class = MinimalCourseRunSerializer
+    TAG = 'test-tag'
```
Comment (clintonb) [inline course_discovery/apps/api/tests/test_serializers.py]: This is not necessary.

File: course_discovery/apps/api/serializers.py [hunk 3]
```diff
@@ -334,6 +336,7 @@ class MinimalCourseRunSerializer(TimestampModelSerializer):
     def prefetch_queryset(cls):
         return CourseRun.objects.all().select_related('course').prefetch_related(
             'course__partner',
+            'tags',
```
Comment (clintonb) [inline course_discovery/apps/api/serializers.py]: This shouldn't be here given that `MinimalCourseRunSerializer` does not serialize tags.

**Review by clintonb — CHANGES_REQUESTED** [review]

File: course_discovery/apps/api/tests/test_serializers.py (hunk 2 above)
Comment (MatthewPiatetsky) [reply course_discovery/apps/api/tests/test_serializers.py]: Are you suggesting just removing the class attribute and hardcoding the string when I add the tag, or are you suggesting creating a tag factory and adding that as a post generation value to the course run factory?     
This is how the Organization tag tests do it, so I'd probably want to change those if I did this differently

Comment (clintonb) [reply course_discovery/apps/api/tests/test_serializers.py]: Neither `MinimalCourseRunSerializer` nor `MinimalCourseRunSerializerTests` deal with tags. Neither class should be modified.

File: course_discovery/apps/api/tests/test_serializers.py (hunk 1 above)
Comment (MatthewPiatetsky) [inline course_discovery/apps/api/tests/test_serializers.py]: removed

File: course_discovery/apps/api/serializers.py (hunk 3 above)
Comment (MatthewPiatetsky) [reply course_discovery/apps/api/serializers.py]: removed

File: course_discovery/apps/api/tests/test_serializers.py (hunk 2 above)
Comment (MatthewPiatetsky) [reply course_discovery/apps/api/tests/test_serializers.py]: but CourseRunSerializerTests also runs the test_data method defined in MinimalCourseRunSerializerTests and CourseRunSerializer does deal with tags

Comment (clintonb) [reply course_discovery/apps/api/tests/test_serializers.py]: Override `test_data` where necessary. There is no need to modify this class.

Comment (clintonb) [reply course_discovery/apps/api/tests/test_serializers.py]: Correction: there is no need to modify this class to make it aware of tags. In lieu of an override, you could also add a method to create the course run, that can also be overridden.

Comment (MatthewPiatetsky) [reply course_discovery/apps/api/tests/test_serializers.py]: got it, added create_course_run method

File: course_discovery/apps/api/tests/test_serializers.py [hunk 4]
```diff
@@ -289,16 +289,21 @@ def get_expected_data(self, course_run, request):  # pylint: disable=unused-argu
             'type': course_run.type,
         }
 
+    def create_course_run(self):
+        course_run = CourseRunFactory()
```
Comment (clintonb) [inline course_discovery/apps/api/tests/test_serializers.py]: `return CourseRunFactory()`

File: course_discovery/apps/api/tests/test_serializers.py [hunk 5]
```diff
@@ -321,13 +326,19 @@ def get_expected_data(self, course_run, request):
             'modified': json_date_format(course_run.modified),  # pylint: disable=no-member
             'level_type': course_run.level_type.name,
             'availability': course_run.availability,
+            'tags': [self.TAG],
         })
 
         return expected
 
+    def create_course_run(self):
+        course_run = CourseRunFactory()
```
Comment (clintonb) [inline course_discovery/apps/api/tests/test_serializers.py]: `course_run = super(CourseRunSerializerTests, self).create_course_run()`

**Review by clintonb — APPROVED** [review]

File: course_discovery/apps/api/v1/tests/test_views/test_course_runs.py [hunk 6]
```diff
@@ -39,7 +39,7 @@ def test_get(self):
         """ Verify the endpoint returns the details for a single course. """
         url = reverse('api:v1:course_run-detail', kwargs={'key': self.course_run.key})
 
-        with self.assertNumQueries(9):
```
Comment (amangano-edx) [inline course_discovery/apps/api/v1/tests/test_views/test_course_runs.py:42]: Why do some of these change by 1 and others by 2?

Comment (MatthewPiatetsky) [reply course_discovery/apps/api/v1/tests/test_views/test_course_runs.py:42]: will fix/explain shortly

File: course_discovery/apps/api/v1/views/catalogs.py [hunk 7]
```diff
@@ -140,8 +140,7 @@ def csv(self, request, id=None):  # pylint: disable=redefined-builtin,unused-arg
 
         # We use select_related and prefetch_related to decrease our database query count
         course_runs = course_runs.select_related(*serializers.SELECT_RELATED_FIELDS['course_run'])
-        prefetch_fields = ['course__' + field for field in serializers.PREFETCH_FIELDS['course']]
```
Comment (MatthewPiatetsky) [inline course_discovery/apps/api/v1/views/catalogs.py:143]: Is this necessary? If so, should we be doing it here?
https://github.com/edx/course-discovery/blob/master/course_discovery/apps/api/v1/views/course_runs.py#L54

**Pull request closed without merging**